### PR TITLE
Cleanup Code & support DecentHolograms

### LINKED ISSUE #48: [Feature] Support Decent Holograms
### What feature do you want to see added?

Support for Decent Holograms 

### Are there any alternatives?

Support for Decent Holograms 
I would like decent hologram support, as the plugin only works with HolographicDisplays

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
Fixes #48

### Code Cleanups
- Remove usage of protocollib - we don't need to spawn individual holograms for each player if we use the Console as the LocaleHolder
- Remove interface for IHoloUtil - there is only one implementation
- Get rid of singleton accessors

### Bugfixes
- Fixes pending holograms on unlink & merge
- Get plot owner using PlayerManager, which utilizes the UUID pipeline (Owner does not neccessarly need to exists in server files and is therefor not present as an OfflinePlayer)
- Don't validate plot boundaries (https://github.com/IntellectualSites/HoloPlots/blob/main/src/main/java/com/plotsquared/holoplots/PSHoloUtil.java#L75) anymore.
  - We take every plot which is connected in any means to the chunk.
  - Fixes an issue where only the sign is exactly in another chunk

### New features
- Multiple providers are now supported
  - New Provider: DecentHolograms

This PR does need quite a bit of testing, especially with multiple accounts - I only have one at the moment (thanks Microsoft).
The core of the plugin has changed fundamentally - so if anything is unclear, I'll glad to explain.

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/main/java/com/plotsquared/holoplots/HoloPlotsPlugin.java [hunk 1]
```diff
@@ -1,35 +1,85 @@
...
+            TranslatableCaption.of("signs.owner_sign_line_3"),
+            TranslatableCaption.of("signs.owner_sign_line_4")
+    );
     private static final int BSTATS_ID = 6402;
-    public static HoloPlotsPlugin THIS;
-    public static IHoloUtil HOLO = null;
+
+    private static final Map<String, Function<HoloPlotsPlugin, IHologramProvider>> SUPPORTED_PROVIDERS = Map.of(
+            "HolographicDisplays", HolographicDisplaysProvider::new,
+            "DecentHolograms", DecentHologramsProvider::new
```
Comment (SirYwell) [inline src/main/java/com/plotsquared/holoplots/HoloPlotsPlugin.java:47]: These Strings are currently duplicated, Can we move them to their corresponding classes as constants?

File: src/main/java/com/plotsquared/holoplots/provider/IHologramProvider.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.bukkit.Location;
+
+import java.util.List;
+
+/**
+ * API for implementing multiple hologram providers and using them with generalized logic.
+ *
+ * @since TODO
+ */
+public interface IHologramProvider {
```
Comment (SirYwell) [inline src/main/java/com/plotsquared/holoplots/provider/IHologramProvider.java:15]: I'd drop the `I` prefix, I don't know why it was used in this project before.

**Review by SirYwell — CHANGES_REQUESTED** [review]
A few minor things, but looks good overall.

**Review by SirYwell — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by NotMyFault** [comment]
Screenshot 2023-01-08 at 17 13 51
After rebooting the server, the head of already claimed plots spawns in the hologram. New plots are unaffected.

**Comment by PierreSchwang** [comment]
> Screenshot 2023-01-08 at 17 13 51 After rebooting the server, the head of already claimed plots spawns in the hologram. New plots are unaffected.

Using DecentHolograms or HolographicDisplays?

**Comment by NotMyFault** [comment]
DecentHolograms

**Comment by PierreSchwang** [comment]
For now blocked by https://github.com/DecentSoftware-eu/DecentHolograms/issues/105

**Comment by PierreSchwang** [comment]
> For now blocked by https://github.com/DecentSoftware-eu/DecentHolograms/issues/105

Was closed as fixed. Haven't tested yet.

**Comment by PierreSchwang** [comment]
Absolute pita to rebase -> #168

**Pull request closed without merging**